### PR TITLE
UPGRADE_NOTES for 0.28 

### PR DESCRIPTION
**Pull request by jwoertink**
## Purpose

UPGRADE_NOTES for Lucky 0.28

### CODE REVIEW

File: UPGRADE_NOTES.md [hunk 1]
```diff
@@ -1,3 +1,87 @@
...
+- Upgrade Lucky CLI (Linux)
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Crystal should be `">= 1.0.0"`
+  - Lucky should be `~> 0.28.0`
+  - Dotenv should be replace with [LuckyEnv 0.1.2](https://github.com/luckyframework/lucky_env)
```
Comment (matthewmcgarvey) [inline UPGRADE_NOTES.md:21]:
```suggestion
  - Dotenv should be replaced with [LuckyEnv 0.1.2](https://github.com/luckyframework/lucky_env)
```

File: UPGRADE_NOTES.md [hunk 2]
```diff
@@ -1,3 +1,87 @@
...
+
+- Run `shards update`
+
+### General updates
+
+- Remove: `needs context : HTTP::Server::Context` from any component, as well as passing it in to the `mount()` for the components. [See PR for details](https://github.com/luckyframework/lucky/pull/1488)
+- Rename: all call to `DeleteOperation.destroy` -> `DeleteOperation.delete`
```
Comment (matthewmcgarvey) [inline UPGRADE_NOTES.md:28]:
```suggestion
- Rename: all `DeleteOperation.destroy` calls to `DeleteOperation.delete`
```

File: UPGRADE_NOTES.md [hunk 3]
```diff
@@ -1,3 +1,89 @@
...
+```
+- Update: your `config/env.cr` to this.
+```crystal
+# Environments are managed using `LuckyEnv`. By default, development, production
+# and test are supported.
+
+# If you need additional environment support, add it here
+# LuckyEnv.add_env :staging
+```
+- Update: any use of `Lucky::Env` to use `LuckyEnv`. (e.g. `Lucky::Env.test?` -> `LuckyEnv.test?`). [See PR for details](https://github.com/luckyframework/lucky_cli/pull/655)
```
Comment (jwoertink) [inline UPGRADE_NOTES.md:60]: Also update any use of `Lucky::Env.name` to use `LuckyEnv.environment`

File: UPGRADE_NOTES.md [hunk 4]
```diff
@@ -1,3 +1,90 @@
...
+
+> Remove the existing Lucky binary and follow the Linux
+> instructions in this section
+> https://luckyframework.org/guides/getting-started/installing#on-linux
+
+- Update versions in `shard.yml`
+  - Crystal should be `">= 1.0.0"`
+  - Lucky should be `~> 0.28.0`
+  - Authentic should be `~> 0.8.0`
+  - Carbon should be `~> 0.2.0`
```
Comment (jwoertink) [inline UPGRADE_NOTES.md:22]: `carbon_sendgrid_adapter ~> 0.1.0` should be included if you're using sendgrid to send emails

File: UPGRADE_NOTES.md [hunk 5]
```diff
@@ -1,3 +1,90 @@
...
+# Environments are managed using `LuckyEnv`. By default, development, production
+# and test are supported.
+
+# If you need additional environment support, add it here
+# LuckyEnv.add_env :staging
+```
+- Update: any use of `Lucky::Env` to use `LuckyEnv`. (e.g. `Lucky::Env.test?` -> `LuckyEnv.test?`). [See PR for details](https://github.com/luckyframework/lucky_cli/pull/655)
+- Update: any use of `Lucky::Env.name` to use `LuckyEnv.environment`.
+- Update: any use of `route` or `nested_route`, and replace them with the generated routes. Use `lucky routes` to view all generated routes. If you still need this, you can use the [Lucky Legacy Routing](https://github.com/matthewmcgarvey/lucky_legacy_routing) shard.
+- Add: the [luckyframework/carbon_sendgrid_adapter](https://github.com/luckyframework/carbon_sendgrid_adapter) shard if you're using Sendgrid to send mail.
```
Comment (jwoertink) [inline UPGRADE_NOTES.md:63]: Be sure to require it in your `config/email.cr` file

**Pull request merged**